File: sonarqube-companion-frontend/src/app/config/model/member.ts
import {MemberData} from '../service/member-data';

export class Member {
  firstName: string;
  lastName: string;
  mail: string;
  uuid: string;
  aliases: string[];
  groups: string[];


  constructor(data: MemberData) {
    this.firstName = data.firstName;
    this.lastName = data.lastName;
    this.mail = data.mail;
    this.uuid = data.uuid;
    this.aliases = data.aliases || [];
    this.groups = data.groups || [];

  }
}

File: sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/config/model/Member.java
package pl.consdata.ico.sqcompanion.config.model;

import lombok.*;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Member {
    private String firstName;
    private String lastName;
    private String mail;
    @EqualsAndHashCode.Include
    private String uuid;
    private Set<String> aliases;
    private Set<String> groups;
    private boolean remote;
    private String remoteType;

}
